static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx

```diff
@@ -336,30 +344,24 @@ function AndroidInstallSteps({
   );
 }
 
-function IOSInstallSteps({
-  project,
-  sdkUpdates,
-}: {
-  project: Project;
-  sdkUpdates: RequestState<ProjectSdkUpdates | null>;
-}) {
+function IOSInstallSteps({project, sdkUpdates, organization}: InstallStepsProps) {
+  const hasSdkUpdates = sdkUpdates.type === 'resolved' && sdkUpdates.data !== null;
   const requiresSdkUpdates =
-    sdkUpdates.type === 'resolved' && sdkUpdates.data?.sdkVersion
+    hasSdkUpdates && sdkUpdates.data?.sdkVersion
       ? semverCompare(sdkUpdates.data.sdkVersion, '7.23.0') < 0
       : false;
-
   return (
     <Fragment>
-      {requiresSdkUpdates ? (
+      {hasSdkUpdates && requiresSdkUpdates && (
         <li>
           <StepTitle>{t('Update your projects SDK version')}</StepTitle>
           <ProjectSdkUpdate
-            minSdkVersion="7.23.0 (sentry-cocoa)"
             project={project}
-            sdkUpdates={sdkUpdates}
+            sdkUpdates={sdkUpdates.data!}
```

Comment:
tried my best to get typescript to infer that `sdkUpdates.data` was not null... 🤷🏼 

![image](https://user-images.githubusercontent.com/7349258/191319550-d3365c1f-2e4e-4d54-991b-56f364400696.png)

Reply:
@eliashussary you need to pull sdkUpdates.data into a variable and do the check on that variable, then TS should infer it correctly

Reply:
you might have to inline the expression for `hasSdkUpdates` for that to work

Reply:
L348 should be sufficient....

this is the only way it'll work; we're doing all the same checks above. its unnecessary to repeat it again. 
![image](https://user-images.githubusercontent.com/7349258/191327040-2b895e56-fcd2-40bf-ba44-985fe8a7406c.png)